chain-crypto/src/algorithms/ed25519_extended.rs

```diff
@@ -75,22 +75,26 @@ mod test {
     use crate::key::KeyPair;
     use crate::sign::test::{keypair_signing_ko, keypair_signing_ok};
 
-    #[quickcheck]
-    fn sign_ok(input: (KeyPair<Ed25519Extended>, Vec<u8>)) -> bool {
-        keypair_signing_ok(input)
+    use proptest::prelude::*;
+    use test_strategy::proptest;
+
+    #[proptest]
+    fn sign_ok(input: (KeyPair<Ed25519Extended>, Vec<u8>)) {
+        prop_assert!(keypair_signing_ok(input))
     }
 
-    #[quickcheck]
-    fn sign_ko(input: (KeyPair<Ed25519Extended>, KeyPair<Ed25519Extended>, Vec<u8>)) -> bool {
-        keypair_signing_ko(input)
+    #[proptest]
+    fn sign_ko(input: (KeyPair<Ed25519Extended>, KeyPair<Ed25519Extended>, Vec<u8>)) {
+        prop_assert!(keypair_signing_ko(input))
     }
 
-    #[quickcheck]
-    /// `secret_from_binary` should fail if the provided byte array does not match the public key size
-    fn secret_from_binary_size_check(n: usize) {
+    // `secret_from_binary` should fail if the provided byte array does not match the public key size
+    // the size is limited to 1 MiB to avoid segfaults during testing
+    #[proptest]
+    fn secret_from_binary_size_check(#[strategy(..1_048_576usize)] n: usize) {
```

Comment:
Can we take the `1_048_576usize` into some const? It is used somewhere else too.